packages/yarnpkg-pnp/sources/esm-loader/loader.ts

```diff
@@ -136,7 +153,41 @@ export async function getSource(urlString: string, context: any, defaultGetSourc
   const url = new URL(urlString);
   if (url.protocol !== `file:`) return defaultGetSource(url, context, defaultGetSource);
 
+  urlString = fileURLToPath(urlString);
+
+  if (realModules.has(urlString)) {
+    return {
+      source: await readFile(urlString),
+    };
+  }
+
+  const fakeModulePath = path.join(path.dirname(urlString), `noop.js`);
+
+  const require = createRequire(fakeModulePath);
+  const dynModule = require(urlString);
+
+  let exportStrings: Array<string> = [];
+  if (dynModule.__esModule === true) {
+    exportStrings = Object.getOwnPropertyNames(dynModule).map(propKey => {
+      if (propKey === `default`) {
+        return `export default cjs['default']`;
+      } else {
+        return `const __${propKey} = cjs['${propKey}'];\n export { __${propKey} as ${propKey} }`;
+      }
+    });
+  } else {
+    exportStrings = [`export default cjs`];
+  }
+
+  const code = `
+  import {createRequire} from 'module';
+  const require = createRequire('${fakeModulePath.replace(/\\/g,`/`)}');
```

Comment:
Probably it's safer to use `createRequire(${JSON.stringify(fakeModulePath)});`, since it automatically handles escapes.